src/Agent.Listener/CommandSettings.cs

```diff
@@ -0,0 +1,237 @@
+        {
+            return GetArgOrPrompt(
+                name: Constants.Agent.CommandLine.Args.Pool,
+                description: StringUtil.Loc("AgentMachinePoolNameLabel"),
+                defaultValue: "default",
+                validator: Validators.NonEmptyValidator);
+        }
+
+        public string GetToken()
+        {
+            return GetArgOrPrompt(
+                name: Constants.Agent.CommandLine.Args.Token,
+                description: StringUtil.Loc("PersonalAccessToken"),
+                defaultValue: string.Empty,
+                validator: Validators.NonEmptyValidator);
+        }
+
+        public string GetUrl()
+        {
+            return GetArgOrPrompt(
+                name: Constants.Agent.CommandLine.Args.Url,
+                description: StringUtil.Loc("ServerUrl"),
+                defaultValue: string.Empty,
+                validator: Validators.ServerUrlValidator);
+        }
+
+        public string GetUserName()
+        {
+            return GetArgOrPrompt(
+                name: Constants.Agent.CommandLine.Args.UserName,
+                description: StringUtil.Loc("UserName"),
+                defaultValue: string.Empty,
+                validator: Validators.NonEmptyValidator);
+        }
+
+        public string GetWindowsLogonAccount(string defaultValue)
+        {
+            return GetArgOrPrompt(
+                name: Constants.Agent.CommandLine.Args.WindowsLogonAccount,
+                description: StringUtil.Loc("WindowsLogonAccountNameDescription"),
+                defaultValue: defaultValue,
+                validator: Validators.NTAccountValidator);
+        }
+
+        public string GetWindowsLogonPassword()
+        {
+            return GetArgOrPrompt(
+                name: Constants.Agent.CommandLine.Args.WindowsLogonAccount,
+                description: StringUtil.Loc("WindowsLogonPasswordDescription"),
+                defaultValue: string.Empty,
+                validator: Validators.NonEmptyValidator);
+        }
+
+        public string GetWork()
+        {
+            return GetArgOrPrompt(
+                name: Constants.Agent.CommandLine.Args.Work,
+                description: StringUtil.Loc("WorkFolderDescription"),
+                defaultValue: "_work",
+                validator: Validators.NonEmptyValidator);
+        }
+
+        //
+        // Private helpers.
+        //
+        private string GetArgOrPrompt(
+            string name,
+            string description,
+            string defaultValue,
+            Func<string, bool> validator)
+        {
+            // Check for the arg in the command line parser.
+            ArgUtil.NotNull(validator, nameof(validator));
+            string result;
+            if (!_parser.Args.TryGetValue(name, out result))
+            {
+                result = null;
+            }
+
+            // Return the arg if it is not empty and is valid.
+            _trace.Info($"Arg '{name}': '{result}'");
```

Comment:
Tracing added into this class so consumers don't need to sprinkle the same monotonous tracing code all over the place.